processing/src/main/java/io/druid/query/aggregation/SimpleDoubleAggregatorFactory.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation;
+
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.Doubles;
+import io.druid.math.expr.ExprMacroTable;
+import io.druid.math.expr.Parser;
+import io.druid.segment.ColumnSelectorFactory;
+import io.druid.segment.DoubleColumnSelector;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+public abstract class SimpleDoubleAggregatorFactory extends AggregatorFactory
+{
+  protected final String name;
+  protected final String fieldName;
+  protected final String expression;
+  protected final ExprMacroTable macroTable;
+
+  public SimpleDoubleAggregatorFactory(
+      ExprMacroTable macroTable,
+      String fieldName,
+      String name,
+      String expression
+  )
+  {
+    this.macroTable = macroTable;
+    this.fieldName = fieldName;
+    this.name = name;
+    this.expression = expression;
+    Preconditions.checkNotNull(name, "Must have a valid, non-null aggregator name");
+    Preconditions.checkArgument(
+        fieldName == null ^ expression == null,
+        "Must have a valid, non-null fieldName or expression"
+    );
+  }
+
+  protected DoubleColumnSelector getDoubleColumnSelector(ColumnSelectorFactory metricFactory, Double nullValue)
```

Comment:
Thanks, but soon with the null support work this will be an object so i will keep it this way.